### ISSUE TITLE
Slurm batch file creation for desi_tile_redshifts on Perlmutter

### ISSUE THREAD

**Issue by marcelo-alvarez** [desihub/desispec#1611]
There is some redundacy in how slurm batch files are created: `desi_proc` uses [`workflow.procfuncs.create_batch_script`](https://github.com/desihub/desispec/blob/master/py/desispec/workflow/procfuncs.py#L300) while `desi_tile_redshifts` uses  [`scripts.tile_redshifts.write_redshift_script`](https://github.com/desihub/desispec/blob/master/py/desispec/scripts/tile_redshifts.py#L243). I have modified the former for usage on Perlmutter, but not the latter, so redshifts batch jobs do not work on Perlmutter at present without modification by hand. 

@akremin and @sbailey please have a look and advise on whether it makes sense to make the same change I made for [`workflow.procfuncs.create_batch_script`](https://github.com/desihub/desispec/blob/master/py/desispec/workflow/procfuncs.py#L300) (2nd to 4th bullets of [#1595](https://github.com/desihub/desispec/pull/1595#issue-1104217793)) to [`scripts.tile_redshifts.write_redshift_script`](https://github.com/desihub/desispec/blob/master/py/desispec/scripts/tile_redshifts.py#L243), or if it's better to refactor this (and possibly other batch automation functionality) so that system-dependency common to all automation of slurm commands and batch files is specified only once.

**Comment by sbailey** [comment]
Indeed, this has evolved organically through multiple generations of the pipeline and has resulted in some duplicated and/or non-coordinated code.

IIUC, the changes needed all apply to the batch script header, and it would be useful to factor that out into a common function that any batch script generation could use.  Please explore that refactor.  Thanks.
